### PR TITLE
fix(j-s): Group sub-article-less laws in legal arguments autofill

### PR DESCRIPTION
**Pull request by oddsson**
## What
Reworks `getLegalArguments` in the indictment count form so that laws without a sub-article (sub-article `0`) are grouped together at the start of the generated legal arguments text, and the remaining laws are grouped by article (paragraph) number.

Previously, a sub-article-less law appearing in the middle of the list produced malformed output (e.g. a stray `og 0.58. gr.` with a glued-together article number). The zero-handling logic only ever applied to the first law in the list, never to later ones.

Also expands the test suite from 7 to 12 cases (covering mid-list zeros, all-zero lists, multiple zeros followed by multiple non-zero groups, a single zero on its own, the general-law-only case, and the empty input), and corrects two inaccurate test descriptions.

## Why
The autofilled "heimfærsla" text was incorrect whenever a sub-article-less law (such as the speeding law `[37, 0]`) appeared anywhere other than first in the list, producing text that prosecutors would have to fix manually. Grouping the sub-article-less laws at the start yields correct, consistent legal arguments.

## Note on production impact (correctness / future-proofing)
This change is about **correctness and future-proofing**, not a bug currently visible in production. Today the only sub-article-less law in `offenseLawsMap` is the speeding law `[37, 0]`, and article `37` is lower than every other article (48, 49, 50, 58, 95). Because the laws are sorted by article number ascending (`getUniqueSortedLaws` / `lawsCompare`) before reaching `getLegalArguments`, `37` already sorts to the front on its own — so with the current data the malformed mid-list ordering can never actually occur.

The new code makes "sub-article-less laws go first" an **explicit rule** rather than an incidental side effect of numeric sorting. This keeps the output correct if a future zero sub-article law with a higher article number is ever added (where sorting alone would otherwise place it in the middle and produce the malformed text). The accompanying tests pass inputs in unsorted order to lock this behavior in regardless of the eventual law set.

## Screenshots / Gifs
Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review
- [x] AI tools were used in preparing this pull request
      Tools used: Claude Code
- [ ] No AI tools were used in preparing this pull request

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved legal-article formatting in indictment counts: correct ordering, grouping, and separator/prefix placement when multiple articles are combined, with special handling for the general article.

* **Tests**
  * Expanded test coverage for legal-argument formatting edge cases, including various zero sub-article scenarios, punctuation/connector placement, and single/multiple article combinations.

### CODE REVIEW

File: apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx [hunk 1]
```diff
@@ -180,42 +180,57 @@ export const getLegalArguments = (
...
-      if (relevantLaws[i - 1][0] !== relevantLaws[i][0]) {
-        andIndex = i
-        break
-      }
+  const generalLaw = hasGeneralLaw
+    ? lawsBroken[lawsBroken.length - 1]
+    : undefined
+  const mainLaws = hasGeneralLaw ? lawsBroken.slice(0, -1) : lawsBroken
+
+  // Laws without a sub-article (sub-article 0) are grouped at the start, the
```
Comment (gudjong) [inline apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx:191]: Is this really what is needed here. The Asana task does not talk about reordering the laws, only about correcting what is displayed

**Review by gudjong — COMMENTED** [review]
One comment.

File: apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx (hunk 1 above)
Comment (oddsson) [reply apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx:191]: Good question. 

This is implemented for correctness and future-proofing but in reality this will never happen given the current laws. There is currently only one law without sub-articles (37) and that law is always rendered first, because it's the lowest number law we currently have (the laws are rendered by ascending order).

I think, if we were to add more laws, it would make sense to group zero sub-article laws together at the start. The logic would then be

> This violates laws A, B and C as well as sub-articles 1 of law D and 2 of law E.

Rather then

> This violates laws A and B as well as sub-article 1 of law D, law C and sub-article 2 of law E.

Comment (gudjong) [reply apps/judicial-system/web/src/routes/Prosecutor/Indictments/Indictment/IndictmentCount.tsx:191]: Maybe, I just don´t know and the ticket is not clear about this. The real question is what the prosecutors want. I will accept, but please consult Guðný about this. Should should have more insight into how this should be.

**Review by gudjong — APPROVED** [review]
Approved with comment.
